platforms/jvm/language-java/src/integTest/groovy/org/gradle/integtests/ExecIntegrationTest.groovy

```diff
@@ -910,6 +911,126 @@ class ExecIntegrationTest extends AbstractIntegrationSpec {
         "javaexec" | { File serverInfoFile -> javaExecSpecWithHttpServer(serverInfoFile) }
     }
 
+    def "execOperations.#method in non-root build.gradle resolves relative path correctly"() {
+        settingsFile << "include 'a'"
+        testDirectory.createDirs("a/build/test/folder")
+        file("a/build.gradle") << """
+            interface ExecOperationsProvider {
+                @Inject
+                abstract ExecOperations getExec()
+            }
+            tasks.register("run") {
+                def execOperations = project.objects.newInstance(ExecOperationsProvider).getExec()
+                doLast {
+                    execOperations.${method} {
+                        $configuration
+                        workingDir = new File("build/test/folder")
```

Comment:
We shouldn't try to support this.  This is always wrong to do.  

We explicitly call this out in our docs:
https://docs.gradle.org/current/userguide/working_with_files.html#sec:single_file_paths

Reply:
The resolution already works without this change also on master, but resolved working dir is inconsistent for different ExecSpec usages, e.g. `Providers.exec` resolves relative path relative to settings dir, while `ExecOperations.exec` resolves it based on Project dir. So this just fixes case for Providers.exec works in Project scope works the same as other ExecSpecs. 

So I think there is no harm in making this consistent.

Additionally this also mimics how it will work with the Provider API, since `DirectoryProperty` does the same as it's implemented here (actually we back workingDir with DirectoryProperty now).

If we don't want that users use relative paths, then we could deprecate passing relative `File` to `DirectoryProperty/RegularFileProperty` in 9.1.